cmd/gomod-zip/zip.go

```diff
@@ -133,6 +133,27 @@ func main() {
 	defer dst.Close()
 	zw := zip.NewWriter(dst)
 
+	haveGoMod := make(map[string]bool)
+	for _, zf := range zr.File {
+		dir, file := path.Split(zf.Name)
+		if file == "go.mod" {
+			haveGoMod[dir] = true
+		}
+	}
+
+	inSubmodule := func(name string) bool {
+		for {
+			dir, _ := path.Split(name)
+			if len(dir) <= len(prefix) {
```

Comment:
dir is a array of strings and prefix is a string ... no? (trying to understand)

Reply:
`dir` is also a string - https://golang.org/pkg/path/#Split

This condition is to ensure we ignore the root go.mod file when searching for submodules.

Reply:
if we only track submodules (see above comment), this could be simplified to `if len(dir) == 0`

Reply:
Done

Reply:
whoops. gotcha!